lib/screens/tripslist.dart

```diff
@@ -19,10 +23,26 @@ class _TripsListState extends State<TripsList> {
   var requestsArray = [];
   @override
   Widget build(BuildContext context) {
+    var _scrollController = ScrollController();
+
     final currentuser = Provider.of<FirebaseUser>(context);
     return StreamBuilder(
-        stream: Firestore.instance.collection('userdetails').document(currentuser.uid).snapshots(),
+        stream: Firestore.instance
+            .collection('userdetails')
+            .document(currentuser.uid)
+            .snapshots(),
         builder: (_, usersnapshot) {
+          _scrollController
+            ..addListener(() {
+              if (_scrollController.position.userScrollDirection.toString() !=
+                  globals.scrollDirection) {
+                setState(() {
+                  globals.scrollDirection =
+                      _scrollController.position.userScrollDirection.toString();
+                });
+                widget.refreshParent();
```

Comment:
@Deepanshu-Rohilla it is only being called when the direction of the scroll changes